include/cppoptlib/linesearch/armijo.h

```diff
@@ -24,6 +24,17 @@ class Armijo {
         const Scalar c = 0.2;
         const Scalar rho = 0.9;
         Scalar alpha = alpha_init;
+        int iter;
+        for (iter = 0; iter < 200; iter++) {
```

Comment:
Why have you gone for this approach instead of just clamping the search point to be within the bounds? This approach is pretty inefficient, and relies on a hardcoded number of iterations (200) which is a bad code smell.